### PR TITLE
Chore(repo): Enable clerk@v5 release process

### PR DESCRIPTION
**Pull request by dimkl**
## Description

To prepare for the next major version (clerk@v5) we need to change our current release process.

Changes:

- [x] Introduced `main-v4` branch to support releasing fixes and changes until next major version is ready for release
- [x] Consider `main-v4` as the stable branch that produces `staging` and `production` releases
- [x] Consider `main` branch as the experimental branch that produces `canary`  releases
- [ ] ~~The `canary` releases broadcast messages to `clerkinc/v5-testbench` that triggers a vercel preview release~~

When next major version is stable and released the `staging` and `production` releases will be produced from the `main` branch. The `main-v4` will NOT be deleted until it's not considered LTS any more.

This PR will be merged after those PRs are being merged: https://github.com/clerkinc/javascript/pulls?q=is%3Apr+is%3Aopen+label%3Av5-pre-release

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [x] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [x] (If applicable) [Documentation](https://github.com/clerkinc/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [x] `build/tooling/chore`

### CODE REVIEW

**Review by nikosdouvlis — COMMENTED** [review]
Could this wait for #1854 to be merged as these scripts have changed a bit?

**Comment by dimkl** [comment]
@nikosdouvlis i will rebase it and have it ready to merge after the https://github.com/clerkinc/javascript/pull/1854

**Review by nikosdouvlis — APPROVED** [review]

**Comment by dimkl** [comment]
Blocked by https://github.com/clerkinc/javascript/pull/1898

**Review by nikosdouvlis — APPROVED** [review]

**Review by desiprisg — APPROVED** [review]

**Pull request merged**
